### ISSUE TITLE
Final link failed.

### ISSUE THREAD

**Issue by so154me** [so154me/stlwrt#1]
I tried compile stlwrt from https://github.com/ohno9119/stlwrt repo and got this error.
```
/usr/bin/ld: src/gtkiconfactory.o: warning: relocation against `gtk_icon_factory_finalize' in read-only section `.text'
/usr/bin/ld: src/gtkaccelmap.o: relocation R_X86_64_PC32 against undefined symbol `gtk_accel_map_init' can not be used when making a shared object; recompile with -fPIC
/usr/bin/ld: final link failed: bad value
```
##### environment setup:
gcc-13.2.0
glibc-2.35
binutils-2.40 compiled with --disable-gold
cflags `-mtune=generic -march=x86-64 -O2 -pipe -fstack-protector -D_FORTIFY_SOURCE=2 -g -fPIC`
ldflags `-Wl,-O1 -Wl,-z,relro -Wl,--hash-style=gnu -Wl,--as-needed -Wl,--sort-common`
ignored_build_types = pae

docker image: https://hub.docker.com/r/uglyside/side/tags

**Comment by so154me** [comment]
@ohno9119

**Comment by so154me** [comment]
@thesquash

**Comment by jcontoso** [comment]
> @ohno9119

Did you use the makefile?
Also the X11 backend is unfinished, so you will still get linking errors if you try to compile that.

**Comment by so154me** [comment]
Yes, I'm using upstream Makefile as is, also I have modified it to use system flags.

**Comment by jcontoso** [comment]
> Yes, I'm using upstream Makefile as is, also I have modified it to use system flags.

Are you sure you are building it as a shared library, do any of your extra cflags conflict with fPIC?

**Comment by so154me** [comment]
Yes.
Full compile log
[stlwrt-build.log](https://github.com/so154me/stlwrt/files/14970987/stlwrt-build.log)

**Comment by jcontoso** [comment]
I tried replicating your build on my system but I do not get this error.
Sorry, there is not much I can help with.

**Comment by so154me** [comment]
Thanks for your answers. I will keep trying. I hope that stlwrt will develop.

This developer is interested in continuing development of GTK2 or forks. https://github.com/wandrien

**Comment by jcontoso** [comment]
> Thanks for your answers. I will keep trying. I hope that stlwrt will develop.
> 
> This developer is interested in continuing development of GTK2 or forks. https://github.com/wandrien

Oh thank you.

**Comment by jcontoso** [comment]
> Thanks for your answers. I will keep trying. I hope that stlwrt will develop.
> 
> This developer is interested in continuing development of GTK2 or forks. https://github.com/wandrien

Do you know anyway I can contact this person directly?

**Comment by so154me** [comment]
I'm not sure I understood your question correctly.

Email address on this page https://wandrien.github.io/about/.

**Comment by ohno9119** [comment]
> I'm not sure I understood your question correctly.
> 
> Email address on this page https://wandrien.github.io/about/.

Btw, are you a friend of Vadim, or otherwise related, just curious.
